doc/source/development/roadmap.rst

```diff
@@ -0,0 +1,145 @@
+.. _roadmap:
+
+=======
+Roadmap
+=======
+
+This page provides an overview of the major themes pandas' development. Implementation
+of these goals may be hastened with dedicated funding.
+
+Extensibility
+-------------
+
+Pandas Extension Arrays provide 3rd party libraries the ability to
+extend pandas' supported types. In theory, these 3rd party types can do
+everything one of pandas. In practice, many places in pandas will unintentionally
+convert the ExtensionArray to a NumPy array of Python objects, causing
+performance issues and the loss of type information. These problems are especially
+pronounced for nested data.
+
+We'd like to improve the handling of extension arrays throughout the library,
+making their behavior more consistent with the handling of NumPy arrays.
+
+String Dtype
+------------
+
+Currently, pandas stores text data in an ``object`` -dtype NumPy array.
+Each array stores Python strings. While pragmatic, since we rely on NumPy
+for storage and Python for string operations, this is memory inefficient
+and slow. We'd like to provide a native string type for pandas.
+
+The most obvious alternative is Apache Arrow. Currently, Arrow provides
+storage for string data. We can work with the Arrow developers to implement
+the operations needed for pandas users (for example, ``Series.str.upper``).
+These operations could be implemented in Numba (
+as prototyped in `Fletcher <https://github.com/xhochy/fletcher>`__)
+or in the Apache Arrow C++ library.
+
+Apache Arrow Interoperability
+-----------------------------
+
+`Apache Arrow <https://arrow.apache.org>`__ is a cross-language development
+platform for in-memory data. The Arrow logical types are closely aligned with
+typical pandas use cases (for example, support for nullable integers).
+
+We'd like have a pandas DataFrame be backed by Arrow memory and data types
+by default. This should simplify pandas internals and ensure more consistent
+handling of data types through operations.
```

Comment:
I am not sure this will *ever* be the default. Sure we can support it thru EA but default is a big ask here.

Reply:
I'll strike "by default"

Reply:
Again, I would also strike the "this should simplify"... again, ``Int64`` already provides the solution to NaNs in ints. I would just say that Arrow is very efficient and portable across languages...

Reply:
Should we remove this entire section?

Reply:
I have no problem with removing it, but I think the following could also work.


```
Apache Arrow Interoperability
-----------------------------

`Apache Arrow <https://arrow.apache.org>`__ is a cross-language development
platform for in-memory data. The Arrow logical types are closely aligned with
typical pandas use cases.

A more integrated support for Arrow memory and data types within DataFrame and
Series will allow users to exploit the performances of such platform, its I/O
capabilities, and to allow for better interoperability with other languages and
libraries supporting it.
```